src/client/Microsoft.Identity.Client/AppConfig/ApplicationOptions.cs

```diff
@@ -125,6 +125,6 @@ public abstract class ApplicationOptions
         /// <summary>
         /// Enables legacy ADAL cache serialization and deserialization.
         /// </summary>
-        public bool LegacyCacheCompatibilityEnabled { get; set; }
+        public bool LegacyCacheCompatibilityEnabled { get; set; } = true;
```

Comment:
Setting the default here also, along with in ApplicationConfiguration class, plus added additional unit tests to test for this.